ITR/data/base_providers.py

```diff
@@ -55,47 +60,61 @@ def _get_projected_production(self, scope: EScope = EScope.S1S2) -> pd.DataFrame
         :param scope: a scope
         :return: pd.DataFrame
         """
-        result = []
-        for bm in self._productions_benchmarks.dict()[str(scope)]['benchmarks']:
-            result.append(self._convert_benchmark_to_series(IBenchmark.parse_obj(bm), scope))
-        df_bm = pd.DataFrame(result)
+        # The call to this function generates a 42-row (and counting...) DataFrame for the one row we're going to end up needing...
+        df_bm = pd.DataFrame([self._convert_benchmark_to_series(bm, scope) for bm in getattr(self._productions_benchmarks, scope.name).benchmarks])
         df_bm.index.names = [self.column_config.REGION, self.column_config.SECTOR, self.column_config.SCOPE]
-
         return df_bm
 
-    def get_company_projected_production(self, company_sector_region_info: pd.DataFrame) -> pd.DataFrame:
-        """
-        get the projected productions for list of companies in ghg_scope12
-        :param company_sector_region_info: DataFrame with at least the following columns :
-        ColumnsConfig.COMPANY_ID, ColumnsConfig.GHG_SCOPE12, ColumnsConfig.SECTOR and ColumnsConfig.REGION
-        :return: DataFrame of projected productions for [base_year - base_year + 50]
-        """
-        benchmark_production_projections = self.get_benchmark_projections(company_sector_region_info, scope=EScope.S1S2)
-        company_production = company_sector_region_info[self.column_config.BASE_YEAR_PRODUCTION]
-        return benchmark_production_projections.add(1).cumprod(axis=1).mul(
-            company_production, axis=0)
-
-    def get_benchmark_projections(self, company_sector_region_info: pd.DataFrame,
-                                  scope: EScope = EScope.S1S2) -> pd.DataFrame:
+    def get_company_projected_production(self, company_sector_region_scope: pd.DataFrame) -> pd.DataFrame:
+        """
+        get the projected productions for list of companies
+        :param company_sector_region_scope: DataFrame with at least the following columns :
+        ColumnsConfig.COMPANY_ID, ColumnsConfig.SECTOR, ColumnsConfig.REGION, ColumnsConfig.SCOPE
+        :return: DataFrame of projected productions for [base_year through 2050]
+        """
+        # get_benchmark_projections is an expensive call.  It's designed to return ALL benchmark info for ANY sector/region combo passed
+        # and it does all that work whether we need all the data or just one row.  Best to lift this out of any inner loop
+        # and use the valuable DataFrame it creates.
+        company_benchmark_projections = self.get_benchmark_projections(company_sector_region_scope)
+        company_production = company_sector_region_scope[[self.column_config.SCOPE, self.column_config.BASE_YEAR_PRODUCTION]].set_index(self.column_config.SCOPE, append=True)
+        company_production = company_production[self.column_config.BASE_YEAR_PRODUCTION]
+        if ITR.isnan(company_production.values[0].m):
+            breakpoint()
```

Comment:
do we want to keep breakpoints in production code?

Reply:
Probably not.  Let me see if I can make a test case trip that, then decide whether the code behaves as expected when so.  It's lazy to leave breakpoints as markers were more testing is needed.

Reply:
I also recall, that you said, that testing is done.
Do we still need more testing in some areas?